### PR TITLE
feat: GKE autopilot support

### PR DESCRIPTION
**Pull request by jmymy**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by jmymy** [comment]
@cmcga1125 can you sign the CLA? If you want credit on the PR. Otherwise I'll have to remove your commits from the PR.

**Comment by jmymy** [comment]
@bharathkkb any insight on what kitchen test is failing?

**Comment by bharathkkb** [comment]
@jmymy It may have been a flake since it was for a different test `verify beta-cluster-local`
```
     ×  Command: `gcloud beta --project=PROJECT_ID container clusters --zone=us-central1 describe simple-regional-beta-cluster-i8nk --format=json` cluster has the expected identityServiceConfig config
     
     expected: {"enabled"=>true}
          got: nil
     
     (compared using ==)
```

**Comment by jmymy** [comment]
> @jmymy It may have been a flake since it was for a different test `verify beta-cluster-local`
> 
> ```
>      ×  Command: `gcloud beta --project=PROJECT_ID container clusters --zone=us-central1 describe simple-regional-beta-cluster-i8nk --format=json` cluster has the expected identityServiceConfig config
>      
>      expected: {"enabled"=>true}
>           got: nil
>      
>      (compared using ==)
> ```

nope that one was me. I found an issue with the merge/rebase and the new identity config.. Should be fixed.

**Comment by jmymy** [comment]
@bharathkkb next steps? I figured i'd give @cmcga1125 more time while ya'll review the PR.

File: examples/simple_autopilot_private/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# Simple Regional Cluster
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/README.md:1]: nit
```suggestion
# Simple Regional Autopilot Cluster
```

File: examples/simple_autopilot_private/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+This example illustrates how to create a simple cluster with beta features.
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/README.md:3]:
```suggestion
This example illustrates how to create a simple autopilot cluster with beta features.
```

File: examples/simple_autopilot_private/variables.tf [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+variable "cluster_name_suffix" {
+  description = "A suffix to append to the default cluster name"
+  default     = ""
+}
+
+variable "region" {
+  description = "The region to host the cluster in"
+}
+
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/variables.tf:29]: We are now trying to hardcode these values in examples as much as possible. So lets remove variables wherever possible

File: examples/simple_autopilot_private/variables.tf [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+variable "compute_engine_service_account" {
+  description = "Service account to associate to the nodes in the cluster"
+}
+
+variable "enable_autopilot" {
+  description = "Enabling autopiolt functionality"
+  default     = true
+}
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/variables.tf:53]: We shouldnt need this

File: examples/simple_autopilot_private/variables.tf [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+  type        = bool
+  description = "(Beta) Enable GKE Sandbox (Do not forget to set `image_type` = `COS_CONTAINERD` and `node_version` = `1.12.7-gke.17` or later to use it)."
+  default     = false
+}
+
+variable "remove_default_node_pool" {
+  type        = bool
+  description = "Remove default node pool while setting up the cluster"
+  default     = false
+}
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/variables.tf:77]: Some of these are not used

File: modules/beta-autopilot-public-cluster/versions.tf [hunk 6]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright 2018 Google LLC
```
Comment (bharathkkb) [inline modules/beta-autopilot-public-cluster/versions.tf:2]: Here and throughout
```suggestion
 * Copyright 2022 Google LLC
```

File: examples/simple_autopilot_private/main.tf [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+data "google_compute_subnetwork" "subnetwork" {
+  name    = var.subnetwork
+  project = var.project_id
+  region  = var.region
+}
+
+module "gke" {
+  source                          = "../../modules/beta-autopilot-private-cluster/"
+  project_id                      = var.project_id
+  name                            = "${local.cluster_type}-cluster${var.cluster_name_suffix}"
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/main.tf:38]:
```suggestion
  name                            = "simple-autopilot-private-cluster${var.cluster_name_suffix}"
```

File: examples/simple_autopilot_private/main.tf [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  host                   = "https://${module.gke.endpoint}"
+  token                  = data.google_client_config.default.access_token
+  cluster_ca_certificate = base64decode(module.gke.ca_certificate)
+}
+
+data "google_compute_subnetwork" "subnetwork" {
+  name    = var.subnetwork
+  project = var.project_id
+  region  = var.region
+}
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/main.tf:33]: Can we create the network in the same example config? We are trying to make examples more standalone and move aways from fixtures https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/930ad85d57e4207360d0833b1b42c8c978d546a4/examples/safer_cluster/network.tf

File: examples/simple_autopilot_private/outputs.tf [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+output "kubernetes_endpoint" {
+  sensitive = true
+  value     = module.gke.endpoint
+}
+
+output "client_token" {
+  sensitive = true
+  value     = base64encode(data.google_client_config.default.access_token)
+}
```
Comment (bharathkkb) [inline examples/simple_autopilot_private/outputs.tf:25]: Lets not output client token unless we are using it in our test. This creates a permadiff for users trying out the example.

**Review by bharathkkb — CHANGES_REQUESTED** [review]
Thanks for working on this @jmymy 
/cc @kaariger since you worked on the KRM autopilot blueprint

**Comment by cmcga1125** [comment]
> @cmcga1125 can you sign the CLA? If you want credit on the PR. Otherwise I'll have to remove your commits from the PR.

Yeah, let me figure out how to do that in the morning

**Comment by jmymy** [comment]
@bharathkkb how do y'all normally handle the version file in `./test/setup` ? its causing the error above.

**Comment by bharathkkb** [comment]
@jmymy I noticed it when working on another PR. You can switch to the registry source https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/99e8f9b88eff8363eb7a06faff79e460c049ac35/test/setup/main.tf

**Comment by jmymy** [comment]
@bharathkkb next steps? Trying to not let this get stale

**Comment by morgante** [comment]
Tests are failing. Please fix integration tests:
```
$$$$$$ Selecting the kitchen-terraform-simple-autopilot-private-local Terraform workspace...
$$$$$$ Finished selecting the kitchen-terraform-simple-autopilot-private-local Terraform workspace.
$$$$$$ Downloading the modules needed for the Terraform configuration...
       - example in ../../../examples/simple_autopilot_private
       Downloading terraform-google-modules/network/google 4.1.0 for example.gcp-network...
       - example.gcp-network in .terraform/modules/example.gcp-network
       - example.gcp-network.firewall_rules in .terraform/modules/example.gcp-network/modules/firewall-rules
       - example.gcp-network.routes in .terraform/modules/example.gcp-network/modules/routes
       - example.gcp-network.subnets in .terraform/modules/example.gcp-network/modules/subnets
       - example.gcp-network.vpc in .terraform/modules/example.gcp-network/modules/vpc
       - example.gke in ../../../modules/beta-autopilot-private-cluster
       Downloading terraform-google-modules/gcloud/google 3.1.0 for example.gke.gcloud_delete_default_kube_dns_configmap...
       - example.gke.gcloud_delete_default_kube_dns_configmap in .terraform/modules/example.gke.gcloud_delete_default_kube_dns_configmap/modules/kubectl-wrapper
       - example.gke.gcloud_delete_default_kube_dns_configmap.gcloud_kubectl in .terraform/modules/example.gke.gcloud_delete_default_kube_dns_configmap
$$$$$$ Finished downloading the modules needed for the Terraform configuration.
$$$$$$ Validating the Terraform configuration files...
       Success! The configuration is valid.
       
$$$$$$ Finished validating the Terraform configuration files.
$$$$$$ Building the infrastructure based on the Terraform configuration...
       
       Error: Invalid for_each argument
       
         on .terraform/modules/example.gcp-network/modules/subnets/main.tf line 29, in resource "google_compute_subnetwork" "subnetwork":
         29:   for_each                 = local.subnets
           ├────────────────
           │ local.subnets will be known only after apply
       
       The "for_each" value depends on resource attributes that cannot be determined
       until apply, so Terraform cannot predict how many instances will be created.
       To work around this, use the -target argument to first apply only the
 ```

**Review by bharathkkb — APPROVED** [review]
Unrelated tests are failing due to flakes

**Pull request merged**